code/game/objects/items/devices/flashlight.dm

```diff
@@ -206,3 +229,79 @@
 
 /obj/item/device/flashlight/slime/attack_self(mob/user)
 	return //Bio-luminescence does not toggle.
+
+//Glowsticks
+
+/obj/item/device/flashlight/glowstick
+	name = "green glowstick"
+	desc = "A green military-grade glowstick."
+	w_class = 2
+	brightness_on = 4

```

Comment:
4 brightness seems a bit much for a glowstick, especially given the duration. i'd say lower it a bit and increase the range a little instead to get a softer light